src/App.tsx

```diff
@@ -1,17 +1,128 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 
 import users from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+import { User, PrepTodo } from './react-app-env';
+
+const getUserById = (userId: number): User | null => {
+  return users.find(user => user.id === userId) || null;
+};
+
+const todosWithUser: PrepTodo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 const App: React.FC = () => {
+  const [todos, setTodos] = useState<PrepTodo[]>(todosWithUser);
+  const [newTodoName, setNewTodoName] = useState('');
+  const [hasNameError, setHasNameError] = useState(false);
+
+  const [selectUserId, setSelectUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const addTodo = (name: string, userId: number) => {
+    const newTodo: PrepTodo = {
+      id: Date.now(),
+      name,
+      userId,
+      completed: false,
```

Comment:
You should add checkbox for completed / non completed value that user can choose in the form